public/app/core/components/AppChrome/TopBar/SingleTopBar.tsx

```diff
@@ -37,7 +38,10 @@ export const SingleTopBar = memo(function SingleTopBar({
   pageNav,
   sectionNav,
 }: Props) {
-  const styles = useStyles2(getStyles);
+  const { chrome } = useGrafana();
+  const state = chrome.useState();
+  const menuDockedAndOpen = !state.chromeless && state.megaMenuDocked && state.megaMenuOpen;
```

Comment:
Just out of curiosity, in the single top bar how is the megamenu in the state docked but not open? 🤔

Reply:
they're separate states. if you dock the menu, then close it, the next time you open the menu it will "remember" that it was docked and open in a docked state. and vice versa.